pkg/squashfs/squashfs.go

```diff
@@ -211,12 +254,28 @@ var squashNotFound = errors.Errorf("squashfuse program not found")
 // return a pointer to the squashfuse cmd.
 // The caller of the this is responsible for the process created.
 func squashFuse(squashFile, extractDir string) (*exec.Cmd, error) {
-	sqfuse := findSquashfusePath()
 	var cmd *exec.Cmd
-	if sqfuse == "" {
+
+	once.Do(findSquashFuseInfo)
+	if squashFuseInfo.Path == "" {
 		return cmd, squashNotFound
 	}
 
+	notifyOpts := ""
+	notifyPath := ""
+	if squashFuseInfo.SupportsNotfiy {
+		sockdir, err := os.MkdirTemp("", "sock")
+		if err != nil {
+			return cmd, err
+		}
+		defer os.RemoveAll(sockdir)
+		notifyPath = filepath.Join(sockdir, "notifypipe")
+		if err := syscall.Mkfifo(notifyPath, 0640); err != nil {
+			return cmd, err
+		}
+		notifyOpts = "notify_pipe=" + notifyPath
+	}
+
 	// given extractDir of path/to/some/dir[/], log to path/to/some/.dir-squashfs.log
 	extractDir = strings.TrimSuffix(extractDir, "/")
```

Comment:
you could put the socket alongside the squashfs log as path/to/some/.dir-squashfuse.socket (and rename the '.dir-squashfs.log' to better '.dir-squash*fuse*.log' at the same time.

Reply:
Well the log is there so it's persistent and you could see what the errors were, I don't think we need persistance for the socket.